src/query/interpreter.cpp

```diff
@@ -1909,6 +1918,119 @@ PreparedQuery PrepareSettingQuery(ParsedQuery parsed_query, const bool in_explic
+
+  Callback callback;
+  switch (transaction_query->action_) {
+    case TransactionQueueQuery::Action::SHOW_TRANSACTIONS: {
+      callback.header = {"username", "transaction_id", "queries summary"};
+      callback.fn = [interpreter_context, username, isAdmin]() mutable {
+        std::vector<std::vector<TypedValue>> results;
+        results.reserve(interpreter_context->interpreters->size());
+        interpreter_context->interpreters.WithLock([&results, username, isAdmin](const auto &interpreters) {
+          spdlog::info("Number of interpreters in callback: {}", interpreters.size());
+          for (Interpreter *interpreter : interpreters) {
+            if (interpreter->HasActiveTransaction() && (interpreter->username_ == username || isAdmin)) {
+              auto queries_summaries_typed = std::vector<TypedValue>();
+              auto queries_summaries = interpreter->GetQueriesSummary();  // in std::transform
+              std::for_each(queries_summaries.begin(), queries_summaries.end(),
+                            [&queries_summaries_typed](const auto &query_summary) {
+                              queries_summaries_typed.emplace_back(query_summary);
+                            });
+
+              results.push_back({TypedValue(interpreter->username_.value_or("")),
+                                 TypedValue(std::to_string(interpreter->GetTransactionId())),
+                                 TypedValue(queries_summaries_typed)});
+            }
+          }
+        });
+        return results;
+      };
+      break;
+    }
+    case TransactionQueueQuery::Action::TERMINATE_TRANSACTIONS: {
+      std::vector<std::string> transaction_ids;
+      std::transform(transaction_query->transaction_id_list_.begin(), transaction_query->transaction_id_list_.end(),
+                     std::back_inserter(transaction_ids), [&evaluator](Expression *expression) {
+                       return std::string(expression->Accept(evaluator).ValueString());
+                     });
+      callback.header = {"transaction_id", "status"};
+      callback.fn = [interpreter_context, db_accessor, transaction_ids, username, isAdmin]() mutable {
+        spdlog::info("Number of transactions to kill in callback: {}", transaction_ids.size());
+        std::vector<std::vector<TypedValue>> results;
+        for (const std::string &transaction_id : transaction_ids) {
+          std::string status{"Transaction not found"};
+          interpreter_context->interpreters.WithLock(
+              [db_accessor, &transaction_id, &status, username, isAdmin](const auto &interpreters) {
+                for (Interpreter *interpreter : interpreters) {
+                  std::string intr_trans_str = std::to_string(interpreter->GetTransactionId());
+                  if (interpreter->HasActiveTransaction() && transaction_id == intr_trans_str) {
+                    if (interpreter->username_ == username || isAdmin) {
```

Comment:
change order maybe:
isAdmin || ...

It seems more likely user will be admin (probably)

Reply:
then in order not to introduce semantic changes, the condition that active transaction exists and is equal has to be duplicated in else branch